### DISCUSSION TITLE
Jellyfin, reverse proxies and web sockets: Why SyncPlay sometimes doesn’t work.

### DISCUSSION THREAD

**Discussion by Moonbase59** [jellyfin/jellyfin#10310] [Show and tell]
People often wonder why (and if) Jellyfin needs _web sockets_ support. YES, it does. Apart from others, this is needed for _SyncPlay_ to function correctly, as I found out the hard way.

I wondered why SyncPlay worked correctly in my local LAN (using [http://media:8096/](http://media:8096/) as its address) _but not_ when using "from outside" via my reverse proxy setup (using Nginx Proxy Manager, and an address like [https://media.example.com/](https://media.example.com/)).

It was a simple solution—enable _Websockets Support_ like so (if using NPM):

![Auswahl_270](https://github.com/jellyfin/jellyfin/assets/3706922/62e8cf9d-567e-40b2-9519-3c16f04adeaf)

Now groups and SyncPlay work fine, even over the Internet and the family can enjoy synchronous playout even when living hundreds of miles apart!

This should probably go into the documentation somewhere, and I leave it here for others who are googling around because they encounter the same problem.

**Comment by Ferk** [comment]
Hi!
Would you mind sharing what the actual nginx configuration file looks like for those of us not using NPM?

I've tried following the instructions [here](https://jellyfin.org/docs/general/networking/nginx/), but it still does not work for me... I think there's an issue with the fact that locally jellyfin is HTTP only while, from the outside, the proxy is HTTPS, so it uses `wss://`. In addition, my jellyfin server uses a subpath (`example.com/media`) instead of a subdomain (`media.example.com/`)...

Below is the relevant section of my config:
```sh
server {
    listen       443 ssl;
   # ... -snip-

    location /media/ {
        proxy_pass http://$nas:8096;
        proxy_pass_request_headers on;

        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Forwarded-Host $http_host;

        # Websocket support
        #proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        #proxy_set_header Connection "upgrade";
        proxy_set_header Connection $http_connection;

        # Disable buffering when the nginx proxy gets very resource heavy upon streaming
        proxy_buffering off;
    }
}
```

**Comment by WeissbrotDE** [comment]
Thank you so much for this!
This is how I fixed it on my OpenMediaVault based Jellyfin server behind an Nginx reverse proxy running on a cloud server:

The important part you need:
```
server {
server_name stream.example.com;
location / {
    proxy_pass http://<internalip>:8096/;
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "Upgrade";
    proxy_set_header Host $host;
}
}
```
Source: [Official Nginx Blog](https://www.nginx.com/blog/websocket-nginx/)


My config (tested):
```
server {
server_name stream.example.com;
location / {
    proxy_pass http://<internalip>:8096/;
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "Upgrade";
    proxy_set_header Host $host;
}


    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/stream.example.com/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/stream.example.com/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

}
server {
    if ($host = stream.example.com) {
        return 301 https://$host$request_uri;
    } # managed by Certbot


server_name stream.example.com;
    listen 80;
    return 404; # managed by Certbot


}
```
⚠️ Remember to replace `<internalip>` with the internal IP of your Jellyfin server and `stream.example.com` with your domain.
